### PR TITLE
Ondrej Michal Add options to dialog

### PR DESCRIPTION
**Pull request by OndraMichal**
US-8506: Web Services: Python: Improvement in initialization of connection with RFEM6

Description:
Purpose of this task is to add new options to input dialogue to enable use already existing models in RFEM or create new one. Edit name of the new model. If existing one is selected then enable deleting results and reset of the model. This envolves changes in initModle.py file to enable loading of existing models befor showing dialogue and then effective use of clientModel ( by creating Model class).

### CODE REVIEW

File: UnitTests/test_basicObjects.py [hunk 1]
```diff
@@ -33,23 +33,23 @@
 
 def test_material():
```
Comment (dogukankaratas) [inline UnitTests/test_basicObjects.py:35]: Model(model_name = 'basicobjects_test')

File: main.py [hunk 2]
```diff
@@ -23,24 +23,23 @@
 from RFEM.Loads.memberLoad import *
 from RFEM.Loads.surfaceLoad import *
 
-def main(hall_width_L, hall_height_h_o, hall_height_h_m, number_frames, frame_spacing):
+def main(hall_width_L, hall_height_h_o, hall_height_h_m, number_frames, frame_spacing, new_model, model_name, delete, reset):
 # -------------------------------------------------------------
-    clientModel.service.begin_modification('new')
-    print('Geometry...')
-
+    Model(hall_width_L, hall_height_h_o, hall_height_h_m, number_frames, frame_spacing, new_model, model_name, delete, reset)
```
Comment (dogukankaratas) [inline main.py:28]: Acc. to previous suggestions, line 28 should be change as;

_Model(new_model, model_name, delete, reset)_

**Review by dogukankaratas — CHANGES_REQUESTED** [review]

File: RFEM/initModel.py [hunk 3]
```diff
@@ -136,18 +134,40 @@
...
+            cModel = Client(new, transport=trans)
+            if delete:
+                print('Deleting results...')
+                cModel.service.delete_all_results()
+            if reset:
+                print('Resetting model...')
+                cModel.service.reset()
+    
+        Model.clientModel = cModel
+        print('Geometry...')
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply RFEM/initModel.py:170]: Well we can print what ever we agree on. By this messages I wanted to comunicate to user, which part of script is running. If Ready seems more appropriate, we can change it. Maybe we should vote on things like that. There is no 100% correct solution.

File: Examples/CantileverQt/Main.py [hunk 4]
```diff
@@ -74,7 +74,7 @@ def onOK(self):
             sys.exit()
 
         # RFEM 6
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/CantileverQt/Main.py:76]: solved

File: Examples/DesignSituations_Test.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/DesignSituations_Test.py:17]: corrected.
Error: Invalid design situation type.' still persist. Out of the scope of this code review.

File: Examples/Free_Load_Test.py [hunk 6]
```diff
@@ -28,7 +28,7 @@
 from RFEM.Loads.surfaceLoad import *
 
 if __name__ == '__main__':
-    clientModel.service.begin_modification('new')
+    Model.clientModel.service.begin_modification('new')
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/Free_Load_Test.py:32]: solved

File: Examples/GlobalParameters_Test.py [hunk 7]
```diff
@@ -13,7 +13,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/GlobalParameters_Test.py:15]: solved
suds.WebFault: Server raised fault: 'Enumeration item 'DEFINITION_TYPE_OPTIMIZATION' not found.'  still persist.

File: Examples/LineLoads_Test.py [hunk 8]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/LineLoads_Test.py:35]: solved. also corrected TabError: inconsistent use of tabs and spaces in indentation

File: Examples/LoadCases_Test.py [hunk 9]
```diff
@@ -29,7 +29,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/LoadCases_Test.py:31]: solved

File: Examples/LoadDistributionSurface_Test.py [hunk 10]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/LoadDistributionSurface_Test.py:35]: solved
suds.TypeNotFound: Type not found: 'load_distribution_direction' still persist.

File: Examples/MemberLoad_test.py [hunk 11]
```diff
@@ -26,7 +26,7 @@
 
 if __name__ == '__main__':
 
-    clientModel.service.begin_modification()
+    Model.clientModel.service.begin_modification()
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/MemberLoad_test.py:30]: solved

File: Examples/MemberSetLoad_test.py [hunk 12]
```diff
@@ -27,7 +27,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/MemberSetLoad_test.py:29]: solved

File: Examples/MembraneSurface_Test.py [hunk 13]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
 
-    clientModel.service.begin_modification('new')
+    Model.clientModel.service.begin_modification('new')
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/MembraneSurface_Test.py:37]: solved

File: Examples/MembraneWithoutTensionSurface_Test.py [hunk 14]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/MembraneWithoutTensionSurface_Test.py:35]: solved
Assert still persist.

File: Examples/NodalLoad_test.py [hunk 15]
```diff
@@ -27,7 +27,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/NodalLoad_test.py:29]: solved.
NameError: name 'ModalSolutionMethod' is not defined still persist.

File: Examples/RigidSurface_Test.py [hunk 16]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/RigidSurface_Test.py:35]: solved

File: Examples/SpectralSettings_test.py [hunk 17]
```diff
@@ -28,7 +28,7 @@
 
 if __name__ == '__main__':
 	#spectral analysis is not yet supported in released RFEM6 in WS
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/SpectralSettings_test.py:30]: solved. suds.TypeNotFound: Type not found: 'ns0:spectral_analysis_settings' still persist.

File: Examples/StandardSurface_Test.py [hunk 18]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/StandardSurface_Test.py:35]: solved

File: Examples/SurfaceLoad_test.py [hunk 19]
```diff
@@ -27,7 +27,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/SurfaceLoad_test.py:29]: solved

File: Examples/SurfaceSetLoad_test.py [hunk 20]
```diff
@@ -28,7 +28,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/SurfaceSetLoad_test.py:30]: solved

File: Examples/Thickness_Test.py [hunk 21]
```diff
@@ -29,7 +29,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/Thickness_Test.py:31]: solved

File: Examples/WithoutThicknessSurface_Test.py [hunk 22]
```diff
@@ -33,7 +33,7 @@
 
 if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/WithoutThicknessSurface_Test.py:35]: solved

File: Examples/modalAnalysis_test.py [hunk 23]
```diff
@@ -27,7 +27,7 @@
 
 if __name__ == '__main__':
     # modal analysis not yet implemmented in released RFEM6
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply Examples/modalAnalysis_test.py:29]: solved. NameError: name 'ModalSolutionMethod' is not defined still persist.

File: RFEM/BasicObjects/material.py [hunk 24]
```diff
@@ -1,14 +1,14 @@
-from RFEM.initModel import *
+from RFEM.initModel import clearAtributes, Model
 
-class Material():
+class Material(Model):
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply RFEM/BasicObjects/material.py:3]: inheritance removed

File: RFEM/initModel.py [hunk 25]
```diff
@@ -136,18 +134,40 @@
...
-    cModel.service.reset()
-else:
-    new = client.service.new_model('My Model') + 'wsdl'
-    cModel = Client(new, transport=trans)
-
-# Init client model
-clientModel = cModel
+class Model():
+    clientModel = None
+    def __init__(self,
```
[the remark this answers is not in the document]
Comment (OndraMichal) [reply RFEM/initModel.py:139]: solved

File: UnitTests/test_basicObjects.py (hunk 1 above)
Comment (OndraMichal) [reply UnitTests/test_basicObjects.py:35]: solved

File: Examples/DesignSituations_Test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dogukankaratas) [reply Examples/DesignSituations_Test.py]: Björn is taking care of that error. @Bjoern-Steinhagen

**Review by dogukankaratas — APPROVED** [review]
- Conflicts should be solven.

**Pull request merged**